plan/logical_plan_builder.go

```diff
@@ -93,6 +88,7 @@ func (b *planBuilder) buildAggregation(p LogicalPlan, aggFuncList []*ast.Aggrega
 	agg.GroupByItems = gbyItems
 	agg.SetSchema(schema)
 	agg.collectGroupByColumns()
+	agg.SetCorrelated()
```

Comment:
All the logical plans calls `SetCorrelated` at the end, can we just use a function to recursively set Correlated of children, just like the physical plan initializer?